source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,169 @@
+  };
+
+  system.stateVersion = "22.05";
+}
+```
+
+## Creating a QEMU based virtual machine using a configuration
+
+A virtual machine is created with the `nix-build` command.
+
+To select `configuration.nix` in the working directory, specify the configuration file as an argument:
+```shell-session
+$ nix-build '<nixpkgs/nixos>' -A vm -I nixos-config=./configuration.nix
+```
+[^nixosrebuild]
+
+This command builds the attribute `vm` utilizing the version of Nixpkgs as specified in the environment variable `NIX_PATH` and using the NixOS configuration as specified in the relative path.
+
+## Running the virtual machine
+
+The previous command creates a link with the name `result` in the working directory.
+It links to the folder that contains the virtual machine.
+
+```shell-session
+$ ls -R ./result
+```
+
+    result:
+    bin  system
+
+    result/bin:
+    run-nixos-vm
+
+
+To run the virtual machine, execute:
+```shell-session
+$ ./result/bin/run-nixos-vm
+```
+
+This command opens a window that shows the boot process of the virtual machine and ends at the `gdm` login screen where you can log in as `alice` with the password `testpw`.
+
+Running the virtual machine will create a nixos.qcow2 file in the folder from which you start the virtual machine.
+This disk image file contains the dynamic state of the virtual machine.
+It can interfere with debugging as it keeps the state of previous runs, for example the user password.
+You should delete this file when you change the configuration.
+
+[^bracket]: As a reminder `configuration.nix` contains a function that returns an [attribute set](https://nixos.org/manual/nix/stable/language/values.html#attribute-set) that follows the convention of a [module](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules).
+In the attribute set you describe how you want your NixOS system configured.
+
+[^password]: Note that this may pose a security risk.
+A more secure option is to encrypt a password using sha-512.
+To create an encrypted password (here using "danger" as an example), execute:
+    ```shell-session
+    $ mkpasswd -m sha-512 danger
+    ```
+
+    Copy the output of `mkpasswd` into the user configuration and use the `initialHashedPassword` option.
+    It should look something like this:
+    ```nix
+        initialHashedPassword = "$6$YLpneQLpnPxXpo1Y$mTZg26KMjWIBqP1N98LzeANb5rfMcC5t7a7Khf/gTB/rPCT4t4x2EgJJZmXkRWcGVW6ZEDMulsjTsXxD7BLZZ/";
+    ```
+
+[^nixpkgs]: Nixpkgs is the largest repository of Nix packages and NixOS modules.
+The repository is hosted on GitHub and maintained by the community, with official backing from the NixOS Foundation.
+
+[^nixosconf]: https://github.com/NixOS/nixpkgs/blob/b4093a24a868708c06d93e9edf13de0b3228b9c7/nixos/modules/installer/tools/tools.nix#L122-L226
+
+[^nixosrebuild]: On NixOS you can create a virtual machine using the command `nixos-rebuild build-vm -I nixos-config=./configuration.nix` which wraps the above command.
```

Comment:
Please move footnotes to [close where they are used](https://github.com/NixOS/nix.dev/blob/master/CONTRIBUTING.md?plain=1#L181-L185). Easier to read and maintain the source.